### PR TITLE
chore: more informative error messages around accepting transactions

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: fedimint-core/src/transaction.rs [hunk 1]
```diff
@@ -125,25 +129,34 @@ pub enum TransactionSignature {
...
+        error: DynInputError,
+        input_idx: u64,
+        txid: TransactionId,
+    },
+    #[error("The transaction {txid} had an invalid output at index {}: {}",  .output_idx, .error)]
+    Output {
+        error: DynOutputError,
+        output_idx: u64,
+        txid: TransactionId,
+    },
```
Comment (elsirion) [inline fedimint-core/src/transaction.rs:161]: Feels like this should be split into a struct containing the txid and an error variant that contains the information currently available. This feels like a useful quick-fix, but not really great code in general.

**Review by elsirion — APPROVED** [review]
I'm on the edge with this one, is there a good reason to repeat the txid over all variants other than "slicghtly easiert to impl"? Feel free to merge to not block #5195 debugging, but I'd rather see it refactored.

**Comment by dpc** [comment]
> I'm on the edge with this one, is there a good reason to repeat the txid over all variants other than "slicghtly easiert to impl"?

If the message is to print the txid (which is handy), then trying to merge it going to be a bit involved. And a bit of very localized redundancy is a minor in-elegancy, not a big deal.

**Comment by dpc** [comment]
This is broken because of #5238

**Comment by dpc** [comment]
Replaced by #5244

**Pull request closed without merging**